src/TasksModule/Components/TasksList/TasksList.tsx

```diff
@@ -1,7 +1,148 @@
-import React from 'react'
+import React, { useEffect, useState } from "react";
+import { baseUrl } from "../../../Constants/Components/Urls.js";
+import axios from "axios";
+import { toast } from "react-toastify";
+import { useNavigate } from "react-router-dom";
+import NoData from "../../../noData/noData.js";
 
 export default function TasksList() {
+  const [taskList, setTaskList] = useState([]);
+  const [titleSearch, setTitleSearch] = useState("");
+  const [statusSearch, setStatusSearch] = useState("");
+  const [pagesArray, setPagesArray] = useState([]);
+
+  const token = localStorage.getItem("userToken");
+  const getTitleValue = (input: any) => {
+    setTitleSearch(input.target.value);
+
+    getTasksList(1, 10, statusSearch, input.target.value);
+  };
+  const getStatusValue = (input: any) => {
+    setStatusSearch(input.target.value);
+    getTasksList(1, 10, input.target.value, titleSearch);
+  };
+  const navigate = useNavigate();
+  const goToAddTasks = () => {
+    navigate("/dashboard/add-tasks");
+  };
+  const getTasksList = async (
+    pageNo: any,
```

Comment:
We should avoid using any as possible and write the correct types for inputs, this will help us detect any potential bugs or errors when we are working.